src/nwbinspector/_formatting.py

```diff
@@ -138,17 +139,17 @@ def _get_message_header(self, message: InspectorMessage):
             message_header += f"with name '{message.object_name}'"
         return message_header
 
-    def _get_message_increment(self, level_counter: list[int]):
+    def _get_message_increment(self, level_counter: list[int]) -> str:
         return (
             f"{'.'.join(np.array(level_counter, dtype=str))}.{self.message_counter}" f"{self.formatter_options.indent}"
         )
 
     def _add_subsection(
         self,
-        organized_messages: dict[str, Union[dict, list[InspectorMessage]]],
+        organized_messages: dict,
```

Comment:
While this could probably be typed with more detail as was attempted originally, it turns out to be pretty tricky to do right so maybe someone can tackle that later on its own PR